### ISSUE TITLE
t.i.task.react's docstring doesn't mention the reactor argument

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6071]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)| @habnabit reported|
|-|-|
|Trac ID|trac#6071|
|Type|defect|
|Created|2012-10-05 21:36:16Z|
|Branch|https://github.com/twisted/twisted/tree/react-reactor-arg-6071|

The docstring for t.i.task.react says:

```
    @param main: A callable which returns a L{Deferred}.  It should take as
        many arguments as there are elements in the list C{argv}.
```

But main is called as:
```finished = main(_reactor, *argv)```

The docstring should be updated to reflect that main will be called with the reactor as an argument.

Attachments:

* [task_docstring.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/524/52461b8a8c7c7577c3bbf0eea6d1c484bfcd59e4/e6d6670f28111725464295d0cfca39224be6328b.patch) (742 bytes) - added by reecer on 2013-04-29 07:57:59Z - Updated the docstring for the PARAM "main" to include that the PARAM _reactor is the first argument when calling "main."
* [react-reactor-arg-6071-v1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/524/52461b8a8c7c7577c3bbf0eea6d1c484bfcd59e4/5a12617d20cfcf5ff2e3b020a2b6feec07ac293e.patch) (1100 bytes) - added by adamtheturtle on 2013-12-07 13:26:51Z - Addresses tomprince's comments
* [react-reactor-arg-6071-v2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/524/52461b8a8c7c7577c3bbf0eea6d1c484bfcd59e4/a1fd78465962b42a4a547b916f9fec54fb651692.patch) (1358 bytes) - added by adamtheturtle on 2013-12-07 13:38:22Z - Like react-reactor-arg-6071-v1.patch, but has a top file

<details><summary>Searchable metadata</summary>

```
trac-id__6071 6071
type__defect defect
reporter__habnabit habnabit
priority__normal normal
milestone__ 
branch__branches_react_reactor_arg_6071 branches/react-reactor-arg-6071
branch_author__reecer reecer
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation_easy documentation easy
time__1349472976000000 1349472976000000
changetime__1386429934000000 1386429934000000
version__None None
owner__rwall rwall

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="reecer's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|reecer commented|
|-|-|

Changed the docstring to the following: 
```
    @param main: A callable which returns a L{Deferred}.  It should take as
        many arguments as there are elements in the list C{argv}. Called with 
        _reactor as first parameter.
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

Thanks for the patch! Make sure you add the "review" keyword when submitting a patch to Twisted, so we know to look at it.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

(In [38461]) Branching to react-reactor-arg-6071.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince commented|
|-|-|

(In [38462]) Apply task_docstring.patch from reecer.

Refs: [#6071](https://github.com/twisted/twisted/issues/6071).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince set owner to reecer|
|-|-|

Thanks for your contribution, and sorry for the delay in responding to it.

Your change is an improvement on the current decription, but it still isn't entirely clear how `main` gets called. It should perhaps say something about taking `reactor`, followed by the elemnts of `argv`.

One thing that the current wording doesn't make clear, is that `main` can have optional and `*args` as well.

(Please submit further patches against the code in the branch)

**Comment by twisted-trac** [comment]
|<img alt="adamtheturtle's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|adamtheturtle removed owner|
|-|-|

Please review react-reactor-arg-6071-v1.patch which adds a clearer explanation of "react"'s main argument, and an example main function.

**Comment by twisted-trac** [comment]
|<img alt="adamtheturtle's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|adamtheturtle commented|
|-|-|

Please review react-reactor-arg-6071-v2.patch, which (on top of the improvements in review react-reactor-arg-6071-v1.patch) has a news file.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_11"></a>|@wallrj commented|
|-|-|

(In [40932]) Apply react-reactor-arg-6071-v2.patch from adamthturtle. Refs [#6071](https://github.com/twisted/twisted/issues/6071)

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_12"></a>|@wallrj set status to `new`|
|-|-|

Thanks Adam,

Looks good. The only thing I notice is that the new example doesn't really belong in the bulleted list.

I'll reformat that and merge.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_13"></a>|@wallrj set status to `closed`|
|-|-|

(In [40934]) Merge react-reactor-arg-6071

Authors: reecer, adamtheturtle
Reviewers: tom.prince, rwall
Fixes: [#6071](https://github.com/twisted/twisted/issues/6071)

The docstring for twisted.internet.task.react now better documents the main
parameter.
